### DISCUSSION TITLE
Does BOINC request a checkpoint when being turned off?

### DISCUSSION THREAD

**Discussion by homersimpsons** [BOINC/boinc#5881] [Q&A] [answered]
I have set checkpoints to happen every 5 minutes of calculations. My CPU computes 20% of the time so it means it will checkpoints every 25 real minutes.

When I turn off my computer would a new checkpoint be requested?

If not I might lose 25 real minutes, maybe I can create an issue for this. If yes, I could not find the answer anywhere so maybe this could be added to a FAQ somewhere?

**Comment by AenBleidd** [comment, marked as answer]
This works different in BOINC.
Basically, we are not asking projects' application to make checkpoint, but rather when the application itself wants to make checkpoint, it asks BOINC client, if it's allowed to checkpoint now.
The reason behind that is because we don't know and can't control when application is able to make checkpoint at all.

**Reply by homersimpsons** [reply]
Using "checkpoint_debug" in event log I notice that some tasks are checkpointing more often than the defined limits. I did set "300s" for this duration, but the logs makes me think that it checkpoints every 30s or so:
![image](https://github.com/user-attachments/assets/df961eee-42ca-4220-988c-2bc325a00979)

**Comment by davidpanderson** [comment]
The 300s is advisory; every 300s the client sends the app a message saying
"checkpoint the next time you can".
The intent is that apps checkpoint only after receiving this message.
But apps are free to checkpoint whenever they want.
Looks like the SiDock app checkpoints frequently.
This is OK as long as the checkpoint files are small.
